src/core/model.jl

```diff
@@ -96,6 +98,39 @@ function Base.show(io::IO, abm::ABM{A}) where {A}
     end
 end
 
+"""
+    agent_validator(agent, space)
+Validate the user supplied agent (subtype of `AbstractAgent`).
+Checks for mutability and existance and correct types for fields depending on `SpaceType`.
+"""
+function agent_validator(::Type{A}, space::S) where {A<:AbstractAgent, S<:SpaceType}
+    # Check A for required properties & fields
+    isbitstype(A) && throw(ArgumentError("Agent struct must be mutable"))
+    any(isequal(:id), fieldnames(A)) || throw(ArgumentError("Agent struct must have an `id` field"))
```

Comment:
I am not sure whether this is entirely true (I don't remember the entire codebase). Let's be safe here and also assert that `:id` not only exists, but it is the first field. (i.e. `fieldnames(A)[1] == :id`.

Reply:
I've added this as an additional check, since we'll get a `BoundsError` if we try this directly with `GridSpace` and `Agent0` for example.